Dockerfile

```diff
@@ -2,23 +2,30 @@ FROM python:3.11-slim AS build
 
 WORKDIR /home/app
 
-COPY requirements/production/requirements.txt .
+RUN pip install poetry
 
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends \
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
         build-essential \
         gcc \
         python3-dev \
-        libpq-dev && \
-    pg_config --version && \
-    python -m venv /home/app/venv && \
-    venv/bin/pip install --no-cache-dir -r requirements.txt
+        libpq-dev
+RUN pg_config --version
+
+# poetry will automatically create a ./.venv/
+RUN poetry config virtualenvs.in-project true
 
+COPY pyproject.toml ./
+COPY poetry.lock ./
+RUN poetry install --only main --no-ansi
 
-FROM python:3.11-slim@sha256:364ee1a9e029fb7b60102ae56ff52153ccc929ceab9aa387402fe738432d24cc
+
+FROM build
```

Comment:
This will bring unnecessary build files into the production image - size increases after this change from 371MB to 1.15GB.

Perhaps pinning to a SHA is unnecessary - I can't remember where I read why to do that.

But this line should at least be `FROM python:3.11-slim AS production` or similar.